fwprovider/acme/datasource_acme_plugin.go

```diff
@@ -0,0 +1,153 @@
+)
+
+// Ensure the implementation satisfies the expected interfaces.
+var (
+	_ datasource.DataSource              = &acmePluginDatasource{}
+	_ datasource.DataSourceWithConfigure = &acmePluginDatasource{}
+)
+
+// NewACMEPluginDataSource is a helper function to simplify the provider implementation.
+func NewACMEPluginDataSource() datasource.DataSource {
+	return &acmePluginDatasource{}
+}
+
+// acmePluginDatasource is the data source implementation for ACME plugin.
+type acmePluginDatasource struct {
+	client *plugins.Client
+}
+
+// Metadata returns the data source type name.
+func (d *acmePluginDatasource) Metadata(
+	_ context.Context,
+	req datasource.MetadataRequest,
+	resp *datasource.MetadataResponse,
+) {
+	resp.TypeName = req.ProviderTypeName + "_acme_plugin"
+}
+
+// Schema returns the schema for the data source.
+func (d *acmePluginDatasource) Schema(
+	_ context.Context,
+	_ datasource.SchemaRequest,
+	resp *datasource.SchemaResponse,
+) {
+	resp.Schema = schema.Schema{
+		Description: "Retrieves a single ACME plugin by plugin ID name.",
+		Attributes: map[string]schema.Attribute{
+			"api": schema.StringAttribute{
+				Description: "API plugin name.",
+				Computed:    true,
+			},
+			"data": schema.MapAttribute{
+				Description: "DNS plugin data.",
+				Computed:    true,
+				ElementType: types.StringType,
+			},
+			"digest": schema.StringAttribute{
+				Description: "Prevent changes if current configuration file has a different digest. " +
+					"This can be used to prevent concurrent modifications.",
+				Computed: true,
+			},
+			"plugin": schema.StringAttribute{
+				Description: "ACME Plugin ID name.",
+				Required:    true,
+			},
+			"type": schema.StringAttribute{
+				Description: "ACME challenge type (dns, standalone).",
+				Computed:    true,
+				Validators: []validator.String{
+					stringvalidator.OneOf("dns", "standalone"),
+				},
+			},
+			"validation_delay": schema.Int64Attribute{
+				Description: "Extra delay in seconds to wait before requesting validation. " +
+					"Allows to cope with a long TTL of DNS records (0 - 172800).",
+				Computed: true,
+				Validators: []validator.Int64{
+					int64validator.Between(0, 172800),
+				},
+			},
+		},
+	}
+}
+
+// Configure adds the provider-configured client to the data source.
+func (d *acmePluginDatasource) Configure(
+	_ context.Context,
+	req datasource.ConfigureRequest,
+	resp *datasource.ConfigureResponse,
+) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	client, ok := req.ProviderData.(proxmox.Client)
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Resource Configure Type",
+			fmt.Sprintf("Expected *proxmox.Client, got: %T",
+				req.ProviderData),
+		)
+
+		return
+	}
+
+	d.client = client.Cluster().ACME().Plugins()
+}
+
+// Read fetches the ACME plugin from the Proxmox cluster.
+func (d *acmePluginDatasource) Read(ctx context.Context, req datasource.ReadRequest, resp *datasource.ReadResponse) {
+	var state acmePluginModel
```

Comment:
Uses the `acmePluginModel` defined in `datasource_acme_plugins.go`.
If preferred, I could extract the model into a separate file.